### PR TITLE
support for changing the actionbar text color

### PR DESCRIPTION
**Pull request by Avinash-Bhat**
The title says all

Other notes:
- This only support setting the color of title (sets `titleTextStyle` in the styles)
- The ABS support may not be correct, as the code is based on the answer at http://stackoverflow.com/a/10357340/1523910
- The Transparent and the Solid title colors are same.

### CODE REVIEW

**Comment by Avinash-Bhat** [comment]
support for changing the tab text color is added in 4222004, though as always, ABS not tested.

**Comment by Avinash-Bhat** [comment]
I have rebased the code onto the latest master after the pull request #89.

**Comment by jgilfelt** [comment]
This looks great. I just need to spend some time testing the various permutations before merging it and deploying. Thanks again!

**Comment by Avinash-Bhat** [comment]
@jgilfelt don't pull this yet, I'm still working on the popup text color and the CAB text color :-)

**Comment by Avinash-Bhat** [comment]
@jgilfelt the support for popup text is somewhat in issue with ABC, in "Light with Dark Actionbar" where the default popup text color is not showing correctly. 

Referring to the code in [android/platform_framework_support](https://github.com/android/platform_frameworks_support/blob/407e01608ccafe5dc24f82608583f71c34312f9c/v7/appcompat/res/values/themes.xml#L144) there were some differences in implementing the styles in our repo, can this be the cause of this issue?

I am not able to get a hold of this.

**Comment by jgilfelt** [comment]
Hmm not sure. I remember a few things being non-trivial with `Light.DarkActionBar` - essentially there is an alternate theme specified in the `actionBarWidgetTheme` attribute whose context is used when inflating widgets into the Action Bar. I think I create a `Theme.<name>.Widget` style for this. Perhaps the popup text styles need setting there?

**Comment by Avinash-Bhat** [comment]
Yup, that fixed it!

**Comment by Avinash-Bhat** [comment]
@jgilfelt I have a question: In the live preview, the bottom bar is the actionbar in contextual mode, isn't it? if so, then changing the "Action mode background color" doesn't change this, only changing the actionbar background color does. Is this a bug?

**Comment by Avinash-Bhat** [comment]
@jgilfelt, never mind I found what it was [here](http://developer.android.com/guide/topics/ui/actionbar.html#SplitBar). 

So there is no preview of CAB style in the UI right?

**Comment by jgilfelt** [comment]
Right, the CAB options were added later and are a minor enough use case that I didn't really wan't to include them in the preview.

With all the additional options, we might want to think about splitting some of the fields (like all the CAB stuff) into an advanced tab or section to simplify things a bit.

**Comment by Avinash-Bhat** [comment]
@jgilfelt I thought so, as trying to support the CAB text style I believe there will be a huge change required, as the simply changing the text color will not be enough, the icons for cut, copy, paste,... should also be changed; maybe some kind of masking also may be required for those to complete.

So I am not going to implement it now -- in this pull request. If you are happy with what is supported now, you can pull this.

**Comment by jgilfelt** [comment]
Yeah that's fine. I never wanted to support even the CAB backgrounds originally because I think the standard colours provide an important visual cue to users that they have entered that mode, so I'm not really bothered about it.

**Comment by Avinash-Bhat** [comment]
@jgilfelt have you been able to test this?

File: src/html/index.html [hunk 1]
```diff
@@ -431,8 +434,11 @@
       
       var DEFAULT_COLORS_DARK = {
       	'backColor':'#2D2D2D',
+        'abTextColor':'#ffffff',
```
Comment (jgilfelt) [inline src/html/index.html:437]: Defaults should correspond with the base holo values. White values should be #f3f3f3

File: src/html/index.html [hunk 2]
```diff
@@ -442,8 +448,11 @@
       
       var DEFAULT_COLORS_INVERSE = {
       	'backColor':'#2D2D2D',
+        'abTextColor':'#ffffff',
```
Comment (jgilfelt) [inline src/html/index.html:451]: Defaults should correspond with the base holo values. White values should be #f3f3f3

File: src/res/actionbar-xml-templates/styles-appcompat.xml [hunk 3]
```diff
@@ -38,16 +41,30 @@
...
+        <item name="titleTextStyle">@style/##stylep##TitleText</item>
     </style>
 
     <style name="ActionBar.Transparent.##stylep##" parent="@style/Widget.AppCompat##base##.ActionBar">
         <item name="background">@drawable/ab_transparent_##style##</item>
         <item name="progressBarStyle">@style/ProgressBar.##stylep##</item>
+        <item name="titleTextStyle">@style/##stylep##TitleText</item>
     </style>
 	
+    <style name="##stylep##TitleText" parent="@style/TextAppearance.AppCompat.Widget.ActionBar.Title">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-appcompat.xml:53]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles-appcompat.xml [hunk 4]
```diff
@@ -38,16 +41,30 @@
...
+        <item name="android:textColor">##actionbar_text_color##</item>
+    </style>
+
     <style name="PopupMenu.##stylep##" parent="@style/Widget.AppCompat##base##.PopupMenu">	
         <item name="android:popupBackground">@drawable/menu_dropdown_panel_##style##</item>	
     </style>
+
+    <style name="##stylep##PopupMenuTextLarge" parent="@style/TextAppearance.AppCompat##base##.Widget.PopupMenu.Large">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-appcompat.xml:61]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles-appcompat.xml [hunk 5]
```diff
@@ -38,16 +41,30 @@
...
+        <item name="android:textColor">##popup_text_color##</item>
+    </style>
+
+    <style name="##stylep##PopupMenuTextSmall" parent="@style/TextAppearance.AppCompat##base##.Widget.PopupMenu.Small">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-appcompat.xml:65]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles-appcompat.xml [hunk 6]
```diff
@@ -56,6 +73,10 @@
     <style name="ActionBarTabStyle.##stylep##" parent="@style/Widget.AppCompat##base##.ActionBar.TabView">
         <item name="android:background">@drawable/tab_indicator_ab_##style##</item>
     </style>
+
+    <style name="##stylep##TabText" parent="@style/Widget.AppCompat.ActionBar.TabText">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-appcompat.xml:77]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles-sherlock.xml [hunk 7]
```diff
@@ -38,16 +41,31 @@
...
+        <item name="titleTextStyle">@style/##stylep##TitleText</item>
     </style>
 
     <style name="ActionBar.Transparent.##stylep##" parent="@style/Widget.Sherlock##base##.ActionBar">
         <item name="background">@drawable/ab_transparent_##style##</item>
         <item name="progressBarStyle">@style/ProgressBar.##stylep##</item>
+        <item name="titleTextStyle">@style/##stylep##TitleText</item>
+    </style>
+    
+    <style name="##stylep##TitleText" parent="@style/TextAppearance.Sherlock.Widget.ActionBar.Title">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-sherlock.xml:53]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles-sherlock.xml [hunk 8]
```diff
@@ -38,16 +41,31 @@
...
+        <item name="android:textColor">##actionbar_text_color##</item>
+        <item name="textColor">##actionbar_text_color##</item>
     </style>
 	
     <style name="PopupMenu.##stylep##" parent="@style/Widget.Sherlock##base##.ListPopupWindow">	
         <item name="android:popupBackground">@drawable/menu_dropdown_panel_##style##</item>	
     </style>
+
+    <style name="##stylep##PopupMenuTextLarge" parent="@style/TextAppearance.Sherlock.Widget.PopupMenu.Large">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-sherlock.xml:62]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles-sherlock.xml [hunk 9]
```diff
@@ -38,16 +41,31 @@
...
+        <item name="android:textColor">##popup_text_color##</item>
+    </style>
+
+    <style name="##stylep##PopupMenuTextSmall" parent="@style/TextAppearance.Sherlock.Widget.PopupMenu.Small">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-sherlock.xml:66]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles-sherlock.xml [hunk 10]
```diff
@@ -56,6 +74,10 @@
     <style name="ActionBarTabStyle.##stylep##" parent="@style/Widget.Sherlock##base##.ActionBar.TabView">
         <item name="android:background">@drawable/tab_indicator_ab_##style##</item>
     </style>
+
+    <style name="##stylep##TabText" parent="@style/Widget.Sherlock##base##.ActionBar.TabText">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-sherlock.xml:78]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles.xml [hunk 11]
```diff
@@ -38,17 +41,31 @@
...
+        <item name="android:titleTextStyle">@style/##stylep##TitleText</item>
     </style>
 
     <style name="ActionBar.Transparent.##stylep##" parent="@android:style/Widget.Holo##base##.ActionBar">
         <item name="android:background">@drawable/ab_transparent_##style##</item>
         <item name="android:progressBarStyle">@style/ProgressBar.##stylep##</item>
+        <item name="android:titleTextStyle">@style/##stylep##TitleText</item>
+    </style>
+
+    <style name="##stylep##TitleText" parent="@android:style/TextAppearance.Widget.ActionBar.Title">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles.xml:53]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles.xml [hunk 12]
```diff
@@ -38,17 +41,31 @@
...
+        <item name="android:textColor">##actionbar_text_color##</item>
     </style>
 
     <style name="PopupMenu.##stylep##" parent="@android:style/Widget.Holo##base##.ListPopupWindow">
         <item name="android:popupBackground">@drawable/menu_dropdown_panel_##style##</item>	
     </style>
 
+    <style name="##stylep##PopupMenuTextLarge" parent="@android:style/TextAppearance.Widget.PopupMenu.Large">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles.xml:61]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles.xml [hunk 13]
```diff
@@ -38,17 +41,31 @@
...
+        <item name="android:textColor">##popup_text_color##</item>
+    </style>
+
+    <style name="##stylep##PopupMenuTextSmall" parent="@android:style/TextAppearance.Widget.PopupMenu.Small">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles.xml:65]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles.xml [hunk 14]
```diff
@@ -57,6 +74,10 @@
         <item name="android:background">@drawable/tab_indicator_ab_##style##</item>
     </style>
 
+    <style name="##stylep##TabText" parent="@android:style/Widget.Holo.ActionBar.TabText">
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles.xml:77]: Need to follow the same convention for the styles previously defined, i.e Name.##stylep##

File: src/res/actionbar-xml-templates/styles.xml (hunk 11 above)
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles.xml:53]: This parent style is invalid - should be TextAppearance.Holo.Widget.ActionBar.Title

File: src/res/actionbar-xml-templates/styles-sherlock.xml [hunk 15]
```diff
@@ -38,16 +41,31 @@
...
 
     <style name="ActionBar.Transparent.##stylep##" parent="@style/Widget.Sherlock##base##.ActionBar">
         <item name="background">@drawable/ab_transparent_##style##</item>
         <item name="progressBarStyle">@style/ProgressBar.##stylep##</item>
+        <item name="titleTextStyle">@style/##stylep##TitleText</item>
+    </style>
+    
+    <style name="##stylep##TitleText" parent="@style/TextAppearance.Sherlock.Widget.ActionBar.Title">
+        <item name="android:textColor">##actionbar_text_color##</item>
+        <item name="textColor">##actionbar_text_color##</item>
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-sherlock.xml:55]: This textColor attribute is invalid and must be removed

File: src/res/actionbar-xml-templates/styles-sherlock.xml [hunk 16]
```diff
@@ -75,6 +97,8 @@
     <style name="Theme.##stylep##.Widget" parent="@style/Theme.Sherlock">
         <item name="popupMenuStyle">@style/PopupMenu.##stylep##</item>
         <item name="dropDownListViewStyle">@style/DropDownListView.##stylep##</item>
+        <item name="android:textAppearanceLargePopupMenu">@style/##stylep##PopupMenuTextLarge</item>
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-sherlock.xml:100]: This attribute is invalid - the android: prefix must be removed

File: src/res/actionbar-xml-templates/styles-sherlock.xml [hunk 17]
```diff
@@ -75,6 +97,8 @@
...
+        <item name="android:textAppearanceSmallPopupMenu">@style/##stylep##PopupMenuTextSmall</item>
```
Comment (jgilfelt) [inline src/res/actionbar-xml-templates/styles-sherlock.xml:101]: This attribute is invalid - the android: prefix must be removed

File: src/html/index.html [hunk 18]
```diff
@@ -950,6 +988,11 @@
             helpText: 'Solid style action bar background.',
             defaultValue: DEFAULT_COLORS_LIGHT['backColor']
           }),
+          new studio.forms.ColorField('abTextColor', {
+            title: 'Action bar text color',
```
Comment (jgilfelt) [inline src/html/index.html:992]: Label is abiguous - should be "Action bar title color" and the help text removed

File: src/html/index.html [hunk 19]
```diff
@@ -959,11 +1002,20 @@
             title: 'Tab indicator color',
             defaultValue: DEFAULT_COLORS_LIGHT['tabColor']
           }),
+          new studio.forms.ColorField('tabTextColor', {
+            title: 'Tab Text color',
```
Comment (jgilfelt) [inline src/html/index.html:1006]: The word "Text" should not be capitalized

File: src/html/index.html [hunk 20]
```diff
@@ -959,11 +1002,20 @@
...
+            defaultValue: DEFAULT_COLORS_LIGHT['tabTextColor']
+          }),
           new studio.forms.ColorField('tertiaryColor', {
             title: 'Popup color',
             helpText: 'Overflow menu, submenu and spinner panel background.',
             defaultValue: DEFAULT_COLORS_LIGHT['tertiaryColor']
           }),
+          new studio.forms.ColorField('popupTextColor', {
+            title: 'Popup text color',
+            helpText: 'Overflow menu, submenu and spinner panel foreground.',
```
Comment (jgilfelt) [inline src/html/index.html:1016]: This help text should be removed

**Comment by jgilfelt** [comment]
Sorry I've been busy with other things and haven't had much time to look at this. Aside from the code comments I've made, the main issue is that the popup text color is not applied to the drop down navigation when the action bar is in list navigation mode. Since the background is applied to both, so should the text color.

**Comment by Avinash-Bhat** [comment]
closing as this is no longer maintained.

**Pull request closed without merging**